packages/stripe-integration/src/stripe-upe/stripe-ocs-payment-strategy.ts

```diff
@@ -0,0 +1,413 @@
+        options: PaymentInitializeOptions & WithStripeUPEPaymentInitializeOptions,
+    ): Promise<void> {
+        const { stripeupe, methodId, gatewayId } = options;
+
+        if (!stripeupe?.containerId) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        if (!gatewayId) {
+            throw new InvalidArgumentError(
+                'Unable to initialize payment because "gatewayId" argument is not provided.',
+            );
+        }
+
+        await this._initializeStripeElement(stripeupe, gatewayId, methodId).catch((error) =>
+            stripeupe.onError?.(error),
+        );
+
+        this.stripeUPEIntegrationService.initCheckoutEventsSubscription(
+            gatewayId,
+            methodId,
+            stripeupe,
+            this.stripeElements,
+        );
+    }
+
+    async execute(orderRequest: OrderRequestBody, options?: PaymentRequestOptions): Promise<void> {
+        const { payment, ...order } = orderRequest;
+        const { methodId, gatewayId } = payment || {};
+
+        if (!this.stripeUPEClient) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        if (!gatewayId || !methodId) {
+            throw new InvalidArgumentError(
+                'Unable to initialize payment because "gatewayId" or "methodId" argument is not provided.',
+            );
+        }
+
+        const state = this.paymentIntegrationService.getState();
+        const { isStoreCreditApplied } = state.getCheckoutOrThrow();
+
+        if (isStoreCreditApplied) {
+            await this.paymentIntegrationService.applyStoreCredit(isStoreCreditApplied);
+        }
+
+        await this.paymentIntegrationService.loadPaymentMethod(gatewayId, {
+            params: { method: methodId },
+        });
+
+        await this.paymentIntegrationService.submitOrder(order, options);
+
+        const paymentMethod = state.getPaymentMethodOrThrow(methodId);
+        const paymentPayload = this._getPaymentPayload(methodId, paymentMethod.clientToken || '');
+
+        try {
+            await this.paymentIntegrationService.submitPayment(paymentPayload);
+        } catch (error) {
+            await this._processAdditionalAction(error, methodId);
+        }
+    }
+
+    finalize(): Promise<void> {
+        return Promise.reject(new OrderFinalizationNotRequiredError());
+    }
+
+    deinitialize(): Promise<void> {
+        this.stripeElements?.getElement(StripeElementType.PAYMENT)?.unmount();
+        this.stripeUPEIntegrationService.deinitialize();
+        this.stripeElements = undefined;
+        this.stripeUPEClient = undefined;
+
+        return Promise.resolve();
+    }
+
+    private async _initializeStripeElement(
+        stripeupe: StripeUPEPaymentInitializeOptions,
+        gatewayId: string,
+        methodId: string,
+    ) {
+        const state = await this.paymentIntegrationService.loadPaymentMethod(gatewayId, {
```

Comment:
Same as above

Reply:
It is the first config load for payment strategy initialisation. We need it to create a Payment Intent (clientToken).